examples/niscope/graph-measurement-betterproto.py

```diff
@@ -181,12 +190,24 @@ async def MeasureGrpcScope(scope_service: niscope_grpc.NiScopeStub, channel, vi)
     except KeyboardInterrupt:
         pass
 
-
+grpc_scope = ()
 async def main():
-    grpc_scope = await OpenGrpcScope(resource, server_address, server_port)
-    await ConfigureGrpcScope(*grpc_scope)
-    await MeasureGrpcScope(*grpc_scope)
-    await CloseGrpcScope(*grpc_scope)
+    try:
+        global grpc_scope
+        grpc_scope = await OpenGrpcScope(resource, server_address, server_port)
+        await ConfigureGrpcScope(*grpc_scope)
+        await MeasureGrpcScope(*grpc_scope)
+    except Exception as e:
+        error_message = e.args[1]
+        if e.args[0] == 22:
+            error_message = f"Failed to connect to server on {server_address}:{server_port}"
+        elif e.status.name == "UNIMPLEMENTED":
+            error_message = "The operation is not implemented or is not supported/enabled in this service"
+        print(error_message)
+    finally:
+        if grpc_scope:
+            await CloseGrpcScope(*grpc_scope)
+        channel.close()
```

Comment:
Here, in the case when server is not running, Channel is still being created but not the session, so `OpenGrpcScope` does not return anything `grpc_scope` tuple. Hence Channel needs to be closed seperately and not as a part of `CloseGrpcScope`